hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -76,6 +77,11 @@ public BaseCommitActionExecutor(HoodieEngineContext context, HoodieWriteConfig c
     // TODO : Remove this once we refactor and move out autoCommit method from here, since the TxnManager is held in {@link AbstractHoodieWriteClient}.
     this.txnManager = new TransactionManager(config, table.getMetaClient().getFs());
     this.lastCompletedTxn = TransactionUtils.getLastCompletedTxnInstantAndMetadata(table.getMetaClient());
+    // TODO: HUDI-2155 bulk insert support bucket index, HUDI-2156 cluster the table with bucket index.
```

Comment:
Can we tackle it in the same PR? bulk. insert is pretty foundational

Reply:
Agree with u, bulk insert is foundational and it has been already implemented. But can we merge this pr first? I consider the bulk insert part will make this pr time period more longer.